app/src/main/java/com/example/meetup/Fragments/SearchFragment.java

```diff
@@ -0,0 +1,391 @@
+import com.google.api.services.language.v1.model.AnalyzeEntitiesResponse;
+import com.google.api.services.language.v1.model.AnnotateTextRequest;
+import com.google.api.services.language.v1.model.Document;
+import com.google.api.services.language.v1.model.Features;
+import com.google.api.services.language.v1.model.Token;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.gson.JsonObject;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class SearchFragment extends Fragment {
+
+    private static final String TAG = "SearchFragment";
+
+    private RecyclerView posts;
+
+    private PostsAdapter adapter;
+    private List<Post> allPosts;
+
+    private AutoCompleteTextView searchName;
+    private AutoCompleteTextView searchCategory;
+    private AutoCompleteTextView searchRole;
+    private AutoCompleteTextView searchKeyWord;
+    private Button searchFind;
+
+    private ArrayList<String> startupNames;
+    private ArrayList<String> categories;
+    private ArrayList<String> roles;
+    private ArrayList<String> keywords;
+
+    private ArrayAdapter<String> searchNameAdapter;
+    private ArrayAdapter<String> searchCategoriesAdapter;
+    private ArrayAdapter<String> searchRolesAdapter;
+    private ArrayAdapter<String> searchKeywordsAdapter;
+
+    private static final int LOADER_ACCESS_TOKEN = 1; // Token used to initiate the request loader
+    private GoogleCredential credential = null; //GoogleCredential object so that the requests for NLP Api could be made
+
+    // A  Thread on which the Api request will be made and results will be delivered. As network calls cannot be made on the amin thread, so we are creating a separate thread for the network calls
+    private Thread thread;
+
+    // Google Request for the NLP Api. This actually is acting like a Http client queue that will process each request and response from the Google Cloud server
+    private final BlockingQueue<CloudNaturalLanguageRequest<? extends GenericJson>> requests
+            = new ArrayBlockingQueue<>(100);
+
+    // Api for CloudNaturalLanguage from the Google Client library, this is the instance of our request that we will make to analyze the text.
+    private CloudNaturalLanguage api = new CloudNaturalLanguage.Builder(
+            new NetHttpTransport(),
+            JacksonFactory.getDefaultInstance(),
+            new HttpRequestInitializer() {
+                @Override
+                public void initialize(HttpRequest request) throws IOException {
+                    credential.initialize(request);
+                }
+            }).build();
+
+
+    public SearchFragment() {
+        // Required empty public constructor
+    }
+
+    interface Function {
```

Comment:
Rename it to QueryResponseCallback
And onCalled should be renamed to onPostsReturned